apiserver/application/application.go

```diff
@@ -69,39 +59,32 @@ func NewFacade(ctx facade.Context) (*API, error) {
 	return NewAPI(
 		backend,
 		ctx.Auth(),
-		ctx.Resources(),
 		ctx.StatePool(),
 		blockChecker,
 		stateCharm,
 		DeployApplication,
-		stateenvirons.GetNewEnvironFunc(environs.New),
 	)
 }
 
 // NewAPI returns a new application API facade.
 func NewAPI(
 	backend Backend,
 	authorizer facade.Authorizer,
-	resources facade.Resources,
 	statePool *state.StatePool,
 	blockChecker BlockChecker,
 	stateCharm func(Charm) *state.Charm,
 	deployApplication func(Backend, jjj.DeployApplicationParams) error,
-	getEnviron stateenvirons.NewEnvironFunc,
 ) (*API, error) {
 	if !authorizer.AuthClient() {
 		return nil, common.ErrPerm
 	}
-	dataDir := resources.Get("dataDir").(common.StringResource)
```

Comment:
Yeah, it takes URLs as args so needs to go onto a controller facade so the proxy can determine where to redirect the api call based on the location of the offers